lib/rubocop/cop/rspec/factory_girl/dynamic_attribute_defined_statically.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryGirl
+        # Prefer to declare dynamic attribute values in a block.
+        #
+        # @example
+        #   # bad
+        #   kind [:active, :rejected].sample
+        #
+        #   # good
+        #   kind { [:active, :rejected].sample }
+        #
+        #   # bad
+        #   closed_at 1.day.from_now
+        #
+        #   # good
+        #   closed_at { 1.day.from_now }
+        #
+        #   # good
+        #   kind :static
+        #
+        #   # good
+        #   comments_count 0
+        #
+        #   # good
+        #   type User::MAGIC
+        class DynamicAttributeDefinedStatically < Cop
+          MSG = 'Use a block to set a dynamic value to an attribute.'.freeze
+
+          def_node_matcher :dynamic_defined_statically?, <<-PATTERN
+          (send nil _ (send ... ))
+          PATTERN
+
+          def_node_search :factory_attributes, <<-PATTERN
+          (block (send nil {:factory, :trait} ...) _ { (begin $...) $(send ...) } )
+          PATTERN
+
+          def on_block(node)
+            return if trait_node?(node)
+            factory_attributes(node).to_a.flatten.each do |attribute|
+              if dynamic_defined_statically?(attribute)
+                add_offense(attribute, :expression)
+              end
+            end
+          end
+
+          def autocorrect(node)
+            lambda do |corrector|
+              expression = node.descendants.first.location.expression
+              corrector.insert_before(expression, '{ ')
```

Comment:
Good question 😄 

I didn't think about this alternative syntax. Do you think there are people declaring factories using parens in syntax?